djautotask/models.py

```diff
@@ -270,7 +271,63 @@ def __str__(self):
 
 
 class LicenseType(Picklist):
-    pass
+    # We expect this id's is fixed and immutable.
+    impersonation_limited_type = {
+        'TEAM_MEMBER': 4,
+        'CONTRACTOR': 8,
+        'CO_HELP_DESK': 9,
+    }
+    impersonation_disabled_type = {
+        'TIME_ATTENDANCE': 5,
+        'DASHBOARD': 6,
+    }
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.init_limited_impersonation_targets()
+
+    def has_impersonation(self, target=None):
+        if self.id in self.impersonation_disabled_type.values():
+            return False
+
+        if self.id in self.impersonation_limited_type.values():
+            return self.check_impersonation(target)
+
+        return True
+
+    def init_limited_impersonation_targets(self):
+
+        if self.id == self.impersonation_limited_type['TEAM_MEMBER']:
+            self.limited_impersonation_targets = [
+                Project,
+                TaskNoteTracker,
+            ]
+        elif self.id == \
+                self.impersonation_limited_type['CONTRACTOR']:
+            self.limited_impersonation_targets = [
+                Ticket,
+                Project,
+                ServiceCallTracker,
+                TimeEntryTracker,
+                TicketNoteTracker,
+                TaskNoteTracker,
+            ]
+        elif self.id == \
+                self.impersonation_limited_type['CO_HELP_DESK']:
+            self.limited_impersonation_targets = [
+                Ticket,
+                Project,
+                TaskNoteTracker,
```

Comment:
This is much better, see how you are able to use classes themselves rather than strings containing method names? This is much cleaner by comparison, and much easier to maintain than having to update a bunch of strings if a method name changes.

Reply:
Yes, I agree. & I am also pleased and happy to see these. Thanks for your valuable review.